spec/features/random_peer_review_spec.rb

```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe 'random peer reviews', type: :feature, js: true do
+  let(:course) { create(:course) }
+  let(:campaign) { create(:campaign) }
+  let(:instructor) { create(:instructor) }
+  let(:student1) { create(:user) }
+  let(:student2) { create(:test_user) }
+
+  before do
+    create(:campaigns_course, campaign_id: campaign.id, course_id: course.id)
+    JoinCourse.new(course: course, user: instructor, role: CoursesUsers::Roles::INSTRUCTOR_ROLE)
+    JoinCourse.new(course: course, user: student1, role: CoursesUsers::Roles::STUDENT_ROLE)
+    JoinCourse.new(course: course, user: student2, role: CoursesUsers::Roles::STUDENT_ROLE)
+
+    # Student1 assigned 2 articles, already reviewing 1 article
+    create(:article, title: 'Football')
+    create(:article, title: 'Basketball')
+    create(:article, title: 'Tennis')
+    create(:assignment, course: course, article_title: Article.first.title, user: student1,
+           role: Assignment::Roles::ASSIGNED_ROLE, wiki: course.home_wiki, article: Article.first)
```

Comment:
Using `Article.first` and the like isn't the best way to do this in a spec. It's essentially a magic number that relies implicitly on the order of creation, so it's hard-coding an invisible bit of coupling.

One better option would be to use `let` to create the articles with specific names.